build/bench/Dockerfile

```diff
@@ -1,59 +1,114 @@
-
-# Set locale en_us.UTF-8 for mariadb and general locale data
-ENV PYTHONIOENCODING=utf-8
-ENV LANGUAGE=en_US.UTF-8
-ENV LANG=en_US.UTF-8
-ENV LC_ALL=en_US.UTF-8
-
-# Install all neccesary packages
-RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-suggests --no-install-recommends \
-  build-essential cron curl git libffi-dev liblcms2-dev libldap2-dev libmariadbclient-dev libsasl2-dev libssl1.0-dev libtiff5-dev \
-  libwebp-dev mariadb-client iputils-ping python3-dev python3-pip python3-setuptools python3-tk redis-tools rlwrap \
-  software-properties-common sudo tk8.6-dev vim xfonts-75dpi xfonts-base wget wkhtmltopdf fonts-cantarell net-tools \
-  && apt-get clean && rm -rf /var/lib/apt/lists/*
-
-# Add frappe user and setup sudo
-RUN groupadd -g 1000 frappe \
-  && useradd -ms /bin/bash -u 1000 -g 1000 -G sudo frappe \
-  && chown -R 1000:1000 /home/frappe \
-  && echo '. "$NVM_DIR/nvm.sh"' >> /home/frappe/.bashrc
-
-# Install nvm with node
-ENV NVM_DIR /home/frappe/.nvm
-ENV NODE_VERSION 12.16.1
-RUN mkdir /home/frappe/.nvm \
-    && wget -qO- https://raw.githubusercontent.com/creationix/nvm/v0.35.2/install.sh | bash \
-    && . $NVM_DIR/nvm.sh \
-    && nvm install $NODE_VERSION \
-    && nvm install 10.19.0 \
-    && npm install yarn -g \
-    && nvm alias default $NODE_VERSION \
-    && nvm use default \
-    && npm install yarn -g \
-    && chown -R frappe:frappe /home/frappe
-ENV NODE_PATH $NVM_DIR/v$NODE_VERSION/lib/node_modules
-ENV PATH $NVM_DIR/v$NODE_VERSION/bin:$PATH
+RUN install_packages \
+    git \
+    wkhtmltopdf \
+    mariadb-client \
+    gettext-base \
+    wget \
+    # for PDF
+    libssl-dev \
+    fonts-cantarell \
+    xfonts-75dpi \
+    xfonts-base \
+    # to work inside the container
+    locales \
+    build-essential \
+    cron \
+    curl \
+    vim \
+    sudo \
+    iputils-ping \
+    watch \
+    tree \
+    nano \
+    software-properties-common \
+    # For psycopg2
+    libpq-dev \
+    build-essential \ 
+    # Other
+    libffi-dev \
+    liblcms2-dev \
+    libldap2-dev \
+    libmariadbclient-dev \
+    libsasl2-dev \
+    libtiff5-dev \
+    libwebp-dev \
+    redis-tools \
+    rlwrap \
+    tk8.6-dev \
+    fonts-cantarell \
+    # VSCode container requirements
+    net-tools \
+    # PYTHON
+    python3-dev \
+    python3-pip \
+    python3-setuptools \
+    python3-tk \
+    python-virtualenv
 
 # Install wkhtmltox correctly
 RUN wget https://github.com/wkhtmltopdf/wkhtmltopdf/releases/download/0.12.5/wkhtmltox_0.12.5-1.stretch_amd64.deb
 RUN dpkg -i wkhtmltox_0.12.5-1.stretch_amd64.deb && rm wkhtmltox_0.12.5-1.stretch_amd64.deb
 
-# Install bench
-RUN pip3 install -e git+https://github.com/frappe/bench.git#egg=bench --no-cache
-
+# Create new user with directory, add user to sudo group, allow passwordless sudo, switch to that user and change directory to user home directory
+RUN useradd --no-log-init -r -m -g  sudo frappe
```

Comment:
why does frappe need sudo?

I removed it and managed to get everything working.

sudo is required mainly to restart supervisor, upgrade letsencrypt. All of that is not required in this docker image. I don't think sudo is required for development

Reply:
In my case I found it necessary if the user wants to install additional packages using apt-get

Reply:
I'll prefer not to have sudo.

If root access is needed. It is container anyway. just `docker exec -u 0 ...`

having sudo will encourage blind use of sudo. "If it doesn't work, use sudo"